### PR TITLE
dev: fix golangci-lint, re-enable depguard, disable nolintlint

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/33183

It also looks like `golangci-lint` has been broken since https://github.com/sourcegraph/sourcegraph/commit/9a4db1085bbf1b86c411975ac87ca3cbc93b2f06 , see https://sourcegraph.slack.com/archives/C01N83PS4TU/p1650565761169409

## Test plan

6642350e1974355a2da15ee980b7a371014818ad

### CODE REVIEW

File: enterprise/dev/ci/internal/ci/operations.go [hunk 1]
```diff
@@ -143,6 +143,7 @@ func addDocs(pipeline *bk.Pipeline) {
 func addCheck(pipeline *bk.Pipeline) {
 	pipeline.AddStep(":clipboard: Misc linters",
 		withYarnCache(),
+		bk.Parallelism(10),
```
Comment (jhchabran) [inline enterprise/dev/ci/internal/ci/operations.go:146]: I assume this is just for testing and will obviously be removed if the test is successful :)

**Review by jhchabran — APPROVED** [review]
Nice to give it another spin! Approving right now to make sure you're not blocked later today :)

File: enterprise/dev/ci/internal/ci/operations.go (hunk 1 above)
Comment (bobheadxi) [reply enterprise/dev/ci/internal/ci/operations.go:146]: Yep!

File: internal/luasandbox/util/util.go [hunk 2]
```diff
@@ -197,13 +197,13 @@ func assertLuaString(value lua.LValue) (string, error) {
 
 // assertLuaNumber returns the given value as a number or an error if the value is
 // of a different type.
-func assertLuaNumber(value lua.LValue) (float64, error) {
-	if value.Type() != lua.LTNumber {
-		return 0, NewTypeError("number", value)
-	}
+// func assertLuaNumber(value lua.LValue) (float64, error) {
```
Comment (bobheadxi) [inline internal/luasandbox/util/util.go:200]: cc @efritz this was reported as unused, can I remove it or should I leave it commented out for now?

Comment (efritz) [reply internal/luasandbox/util/util.go:200]: Eh if we haven't used it yet you can remove it. It's easy to put back in.

File: internal/codeintel/dependencies/live/init.go [hunk 3]
```diff
@@ -26,11 +26,7 @@ func GetService(db database.DB, syncer dependencies.Syncer) *dependencies.Servic
 
 // TestService creates a fresh dependencies service with the given database handle and syncer
 // instance. If the given syncer is nil, then ErrorSyncer will be used instead.
-func TestService(db database.DB, syncer dependencies.Syncer) *dependencies.Service {
-	if syncer == nil {
-		syncer = ErrorSyncer
-	}
-
+func TestService(db database.DB, _ dependencies.Syncer) *dependencies.Service {
 	return dependencies.TestService(db, NewGitService(db), &errorSyncer{})
```
Comment (efritz) [inline internal/codeintel/dependencies/live/init.go:30]:
```suggestion
	return dependencies.TestService(db, NewGitService(db), syncer)
```

Comment (efritz) [reply internal/codeintel/dependencies/live/init.go:30]: YOLO BABY

**Review by camdencheek — APPROVED** [review]

File: dev/check/go-lint.sh [hunk 4]
```diff
@@ -31,27 +31,28 @@ run() {
...
 declare -A IGNORED_DIRS=(
-    ["./docker-images/syntax-highlighter"]=1
+  ["./docker-images/syntax-highlighter"]=1
+  ["./internal/codeintel/dependencies/internal/lockfiles/testdata/parse"]=1
 )
 
 # If no args are given, traverse through each project with a `go.mod`
 if [ $# -eq 0 ]; then
-  find . -name go.mod -type file -exec dirname '{}' \; | while read -r d; do
+  find . -name go.mod -exec dirname '{}' \; | while read -r d; do
```
Comment (keegancsmith) [inline dev/check/go-lint.sh:51]: @bobheadxi this type file was needed. The problem was type file works on mac but not on linux. The portable version was `-type f`. Fix here https://github.com/sourcegraph/sourcegraph/pull/34325

```shellsession
$ diff <(find . -name go.mod -type f) <(find . -name go.mod)
1a2
> ./internal/codeintel/dependencies/internal/lockfiles/testdata/parse/go.mod
```

**Pull request merged**